src/org/ods/orchestration/usecase/JiraUseCase.groovy

```diff
@@ -464,4 +474,131 @@ class JiraUseCase {
+        if (securityVulnerabilityIssues?.size() >= 1) { // Transition the issue to "TO DO" state
+            transitionIssueToToDo(securityVulnerabilityIssues.get(0).id)
+            return (securityVulnerabilityIssues.get(0) as Map)?.key
+        } else { // Create the issue
+            return (createIssueTypeSecurityVulnerability(fixVersion: fixVersion, component: fullJiraComponentName,
+                priority: SECURITY_VULNERABILITY_ISSUE_PRIORITY, projectKey: project.jiraProjectKey,
+                summary: issueSummary, description: description)
+                as Map)?.key
+        }
+    }
+
+    String buildSecurityVulnerabilityIssueDescription(Map vulnerability, String gitUrl, String gitBranch,
+                                                      String repoName, String nexusReportLink) {
+        StringBuilder message = new StringBuilder()
+        message.append("\nAqua security scan detected the remotely exploitable critical " +
+            "vulnerability with name *${vulnerability.name as String}* in repository *[${repoName}|${gitUrl}]* " +
+            "in branch *${gitBranch}*." )
+        message.append("\n\n*Description:* " + vulnerability.description as String)
+        message.append("\n\n*Solution:* " + vulnerability.solution as String)
+
+        if (nexusReportLink != null) {
+            message.append("\n\nYou can find the complete security scan report *[here|${nexusReportLink}]*.")
+        }
+
+        return message.toString()
+    }
+
+    Map createIssueTypeSecurityVulnerability(Map args) {
+        return jira?.createIssue(fixVersion: args.fixVersion, component: args.component,
+            priority: args.priority, summary: args.summary, type: "Security Vulnerability",
+            projectKey: args.projectKey, description: args.description)
+    }
+
+    void transitionIssueToToDo(String issueId) {
+        int maxAttemps = 10;
+        while (maxAttemps-- > 0) {
+            Map response = jira?.getIssueStatusWithTransitions(issueId)
+            String issueStatus = response.status
+            Map possibleTransitionsByName = response.transitions.
+                collectEntries { t -> [t.name.toString().toLowerCase(), t] }
+            if (issueStatus.equalsIgnoreCase("to do")) { // Issue is already in TO DO state
+                return
+            } else if (possibleTransitionsByName.containsKey("implement")) { // We need to transition the issue
+                jira?.doTransition(issueId, possibleTransitionsByName.get("implement"))
+            } else if (possibleTransitionsByName.containsKey("confirm dod")) { // We need to transition the issue
+                jira?.doTransition(issueId, possibleTransitionsByName.get("confirm dod"))
+            } else if (possibleTransitionsByName.containsKey("reopen")) { // We need just one transiton
+                jira?.doTransition(issueId, possibleTransitionsByName.get("reopen"))
+                return
```

Comment:
Remove this return. In next iterarion the code will check if the status is "to do"

Reply:
Agreed.